scripts/irods/test/test_ichksum.py

```diff
@@ -466,3 +467,81 @@ def test_ichksum_detects_when_the_size_in_storage_is_less_than_the_size_in_catal
         # Show that ichksum detects the size inconsistency between the catalog and the object in storage.
         self.admin.assert_icommand(['ichksum', '-f', data_object], 'STDERR', ['-512000 UNIX_FILE_READ_ERR'])
 
+    def test_ichksum_server_honors_server_config_option_checksum_read_buffer_size_in_bytes__issue_7947(self):
+
+        def get_checksum(data_object, replica_number, expected_length_of_checksum=49):
+            gql = "select DATA_CHECKSUM where COLL_NAME = '{0}' and DATA_NAME = '{1}' and DATA_REPL_NUM = '{2}'"
+            checksum, err, ec = self.admin.run_icommand(['iquest', '%s', gql.format(self.admin.session_collection, data_object, replica_number)])
+            self.assertEqual(ec, 0)
+            self.assertEqual(len(err), 0)
+            self.assertEqual(len(checksum.strip()), expected_length_of_checksum)
+            return checksum.strip()
+
+        object_name = 'test_ichksum_server_honors_server_config_option_checksum_read_buffer_size_in_bytes__issue_7947'
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        # update the checksum_read_buffer_size_in_bytes to 10 MiB
+        svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = 10*1024*1024
+
+        # dump to a string to repave the existing server_config.json
+        new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+        try:
+            # create a 100 MiB file
+            file_size = 100*1024*1024
+            local_file = os.path.join(self.admin.local_session_dir, object_name)
+            lib.make_file(local_file, file_size)
+            file_checksum = lib.file_digest(local_file, 'sha256', encoding='base64')
+
+            with lib.file_backed_up(server_config_filename):
+                # repave the existing server_config.json
+                with open(server_config_filename, 'w') as f:
+                    f.write(new_server_config)
+
+                # put the file with new server configuration, run ichksum, and compare checksums
+                self.admin.assert_icommand(['iput', local_file, object_name])
+                self.admin.assert_icommand(['ichksum', '-f', object_name], 'STDOUT_SINGLELINE', file_checksum)
```

Comment:
Is the `-f` needed here in order to register the checksum?

Reply:
I believe so.